src/ncp/spinel.c

```diff
@@ -332,7 +331,14 @@ spinel_datatype_vunpack_(const uint8_t *data_ptr, spinel_size_t data_len, const
 
             if (arg_ptr)
             {
-                *arg_ptr = (spinel_ipv6addr_t *)data_ptr;
+                if (*arg_ptr)
+                {
+                    memcpy(*arg_ptr, data_ptr, sizeof(spinel_ipv6addr_t));
+                }
+                else
+                {
+                    *arg_ptr = (spinel_ipv6addr_t *)data_ptr;
+                }
```

Comment:
Thanks for the explanation, I agree that we should minimize the impact on other projects and lower the possibility of misuse.

For minimizing the impact, I think we can keep the behaviors unchanged for existing APIs. For the misuse concern, I'd like to provide a new C API to support this in-place with well documented notes about the usage. The new API and existing APIs would share most code, so that we don't bloat the code size. Thoughts?